android/java/org/chromium/chrome/browser/crypto_wallet/permission/BraveEthereumPermissionAccountsListAdapter.java

```diff
@@ -50,34 +67,104 @@ public BraveEthereumPermissionAccountsListAdapter(AccountInfo[] accountInfo) {
         mContext = parent.getContext();
         LayoutInflater inflater = LayoutInflater.from(mContext);
         View view = inflater.inflate(R.layout.brave_wallet_accounts_list_item, parent, false);
+        if (!mCheckBoxStyle && mDelegate != null) {
+            mAccountsWithPermissions = mDelegate.getAccountsWithPermissions();
+            mSelectedAccount = mDelegate.getSelectedAccount();
+        }
+
         return new ViewHolder(view);
     }
 
+    public void setAccounts(AccountInfo[] accountInfo) {
+        mAccountInfo = accountInfo;
+    }
+
+    public void setAccountsWithPermissions(HashSet<AccountInfo> accountsWithPermissions) {
+        mAccountsWithPermissions = accountsWithPermissions;
+    }
+
+    public void setSelectedAccount(String selectedAccount) {
+        mSelectedAccount = selectedAccount;
+    }
+
     @Override
     public void onBindViewHolder(
             @NonNull BraveEthereumPermissionAccountsListAdapter.ViewHolder holder, int position) {
         final int arrayPosition = position;
         holder.titleText.setText(mAccountInfo[arrayPosition].name);
         holder.subTitleText.setText(stripAccountAddress(mAccountInfo[arrayPosition].address));
         setBlockiesBitmapResource(holder.iconImg, mAccountInfo[arrayPosition].address);
-        holder.accountCheck.setOnCheckedChangeListener(
-                new CompoundButton.OnCheckedChangeListener() {
-                    @Override
-                    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
-                        if (isChecked) {
-                            mCheckedPositions.add((Integer) arrayPosition);
-                        } else {
-                            mCheckedPositions.remove((Integer) arrayPosition);
+        if (mCheckBoxStyle) {
+            holder.accountAction.setVisibility(View.GONE);
+            holder.accountCheck.setVisibility(View.VISIBLE);
+            holder.accountCheck.setOnCheckedChangeListener(
+                    new CompoundButton.OnCheckedChangeListener() {
+                        @Override
+                        public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                            if (isChecked) {
+                                mCheckedPositions.add((Integer) arrayPosition);
+                            } else {
+                                mCheckedPositions.remove((Integer) arrayPosition);
+                            }
                         }
+                    });
+        } else {
+            holder.accountCheck.setVisibility(View.GONE);
```

Comment:
nit: accountCheck visibility is already GONE in xml

Reply:
done